sampler.go

```diff
@@ -388,34 +403,48 @@ func (s *httpSamplingManager) GetSamplingStrategy(serviceName string) (*sampling
 // the sampling strategy from an HTTP sampling server (e.g. jaeger-agent).
 func NewRemotelyControlledSampler(
 	serviceName string,
-	initial Sampler,
-	hostPort string,
-	metrics *Metrics,
-	logger Logger,
+	options ...SamplerOption,
 ) *RemotelyControlledSampler {
-	if hostPort == "" {
-		hostPort = defaultSamplingServerHostPort
-	}
-	manager := &httpSamplingManager{serverURL: "http://" + hostPort}
-	if logger == nil {
-		logger = NullLogger
-	}
+	initialSampler, _ := NewProbabilisticSampler(0.001)
 	sampler := &RemotelyControlledSampler{
-		serviceName: serviceName,
-		manager:     manager,
-		logger:      logger,
-		metrics:     *metrics,
-		timer:       time.NewTicker(1 * time.Minute),
-	}
-	if initial != nil {
-		sampler.sampler = initial
-	} else {
-		sampler.sampler, _ = NewProbabilisticSampler(0.001)
+		serviceName:   serviceName,
+		logger:        NullLogger,
+		metrics:       NewMetrics(NullStatsReporter, nil),
+		timer:         time.NewTicker(1 * time.Minute),
+		hostPort:      defaultSamplingServerHostPort,
+		sampler:       initialSampler,
+		maxOperations: defaultMaxOperations,
 	}
+
+	sampler.applyOptions(options...)
+	sampler.manager = &httpSamplingManager{serverURL: "http://" + sampler.hostPort}
+
 	go sampler.pollController()
 	return sampler
 }
 
+func (s *RemotelyControlledSampler) applyOptions(options ...SamplerOption) {
```

Comment:
I'm really not a fan of having opts as one of the variables inside the remotecontrolledsampler struct so I went with this approach. However, this might get unwieldy if more options are added...